File: lab7/zeta.cpp
#include <iostream>
#include <cstdlib>
#include <cmath>
using namespace std;


void findZetaSum(int x, int n)
{
//	cout<<"inside zetaSum function"<<endl;
//	cout<<"x is"<<x<<endl;
//	cout<<"n is"<<n<<endl;
        double answer = 0;
        for(int i = 1; i<= n; i++)
        {
//                cout<< answer <<endl;
		double frac = 1/((double)i);
//		cout<<"frac: "<<frac<<" i:"<<i<<"x: "<<x<<endl;
		double addition = pow(frac, x);
//		cout<<"addition:"<<addition<<endl;
                answer = answer + addition;
        }
        cout<<"Z("<<x<<") ="<<answer<<" when approximated at n= "<<n<<endl;

}
    
int main(int argc, char *argv[]){
	findZetaSum(atoi(argv[1]), atoi(argv[2]));
}               

File: lab9/zeta.java
public class zeta
{
    public static void main(String[] args)
    {
        //System.out.println("Your first argument is: "+args[0]);
        //System.out.print("hello");
        findZetaSum(Integer.parseInt(args[0]), Integer.parseInt(args[1]));
    }

    public static void findZetaSum(int x, int n)
    {
        double answer = 0;
        for (int i = 1; i <= n; i++)
        {
            double frac = 1 / ((double)i);
            double addition = Math.pow(frac, x);
            answer = answer + addition;
        }
        System.out.print("Z(");
        System.out.print(x);
        System.out.print(") =");
        System.out.print(answer);
        System.out.print(" when approximated at n= ");
        System.out.print(n);
        System.out.print("\n");
    }
}
